diff --git forkSrcPrefix/source/textures.c forkDstPrefix/source/textures.c
index fb38b639380a2091689e4d83446c1da8e7a07a0a..aad24a9a29a287841bb77988ec4cc90d798b2d8f 100644
--- forkSrcPrefix/source/textures.c
+++ forkDstPrefix/source/textures.c
@@ -613,7 +613,11 @@ static inline __attribute__((always_inline)) void _glTexImage2D_FlatIMPL(texture
 		switch (type) {
 		case GL_UNSIGNED_BYTE:
 			data_bpp = 1;
-			if (internalFormat == GL_LUMINANCE || internalFormat == GL_SLUMINANCE || internalFormat == GL_SLUMINANCE8)
+			if (internalFormat == GL_LUMINANCE 
+				|| internalFormat == GL_SLUMINANCE
+				|| internalFormat == GL_SLUMINANCE8
+				// Hack for BGDA textures: added this condition to force fast path
+				|| internalFormat == GL_COLOR_INDEX8_EXT)
 				fast_store = GL_TRUE;
 			else
 				read_cb = readL;
@@ -825,7 +829,8 @@ static inline __attribute__((always_inline)) void _glTexImage2D_FlatIMPL(texture
 		break;
 	case GL_COLOR_INDEX8_EXT:
 		tex->write_cb = writeR; // TODO: This is a hack
-		tex_format = SCE_GXM_TEXTURE_FORMAT_P8_ABGR;
+		// Hack specific for BGDA texture format
+		tex_format = SCE_GXM_TEXTURE_FORMAT_P8_ARGB;
 		break;
 	case GL_SLUMINANCE:
 	case GL_SLUMINANCE8:
@@ -1771,7 +1776,16 @@ void glDeleteTextures(GLsizei n, const GLuint *gl_textures) {
 						texture_slots[i].dirty = GL_TRUE;
 					}
 				else
+				{
+					SceGxmTexture* gxmTex = &texture_slots[i].gxm_tex;
+					void* paletteData = sceGxmTextureGetPalette(gxmTex);
+					if (paletteData)
+					{
+						vgl_free(paletteData);
+						texture_slots[i].palette_data = NULL;
+					}
 					gpu_free_texture(&texture_slots[i]);
+				}
 				break;
 			case TEX_UNINITIALIZED:
 				texture_slots[i].status = TEX_UNUSED;
diff --git forkSrcPrefix/source/lookup.c forkDstPrefix/source/lookup.c
index 1264dd2a2039eac593d132b07836e0407a9a5fce..797f1f3ae9adbe447a3e86d5cc033556764b53a4 100644
--- forkSrcPrefix/source/lookup.c
+++ forkDstPrefix/source/lookup.c
@@ -441,6 +441,7 @@ static const struct {
 	{"vglUseVramForUSSE", (void *)vglUseVramForUSSE},
 	{"vglUseExtraMem", (void *)vglUseExtraMem},
 	{"vglWaitVblankStart", (void *)vglWaitVblankStart},
+	{"vglSetTexPalette", (void*)vglSetTexPalette}
 #endif
 	// *egl
 	{"eglBindAPI", (void *)eglBindAPI},
@@ -470,10 +471,22 @@ void *vglGetProcAddress(const char *name) {
 	const int len = strlen(name);
 	char tmpname[len + 1];
 	vgl_fast_memcpy(tmpname, name, len + 1);
-	if (!strcmp(tmpname + len - 3, "EXT") || !strcmp(tmpname + len - 3, "ARB") || !strcmp(tmpname + len - 3, "OES")) {
+	if (!strcmp(tmpname + len - 3, "EXT") 
+	|| !strcmp(tmpname + len - 3, "ARB") 
+	|| !strcmp(tmpname + len - 3, "OES")) {
 		tmpname[len - 3] = 0;
 	}
 
+	// strip NV extension
+	if (!strcmp(tmpname + len - 2, "NV")) {
+		tmpname[len - 2] = 0;
+	}
+
+	// strip QCOM extension
+	if (!strcmp(tmpname + len - 4, "QCOM")) {
+		tmpname[len - 4] = 0;
+	}
+
 	// search for stripped name
 	for (size_t i = 0; i < vgl_numproc; ++i) {
 		if (!strcmp(tmpname, vgl_proctable[i].name)) {
